sa/sa.go

```diff
@@ -698,7 +698,7 @@ func (ssa *SQLStorageAuthority) CountFQDNSets(ctx context.Context, req *sapb.Cou
 
 // FQDNSetTimestampsForWindow returns the issuance timestamps for each
 // certificate, issued for a set of domains, during a given window of time, in
-// ascending order.
+// descending order.
```

Comment:
I'd explicitly spell out "most recent issuance first".

Reply:
Why switch from ascending to descending? Ascending seems like the natural order (going forward in time); it just means the receiving code has to iterate through the timestamps backwards.

Also it's worth pointing out that MariaDB indexes have an inherent order, which defaults to ASC. So for instance our

```sql
  KEY `setHash_issued_idx` (`setHash`,`issued`)
```

Is in ASC order. Presumably it's pretty easy for MariaDB to figure things out and traverse the index in reverse, but with more complicated indexes I think it starts to matter more whether your sort order is the same as your index order.

Reply:
Bump on this question

Reply:
> Bump on this question

Ah, my apologies, I didn't see this here when I marked the conversation resolved.

I originally wrote a version of this which iterated over the results in reverse order. It felt messier than adjusting the query to use the descending `ORDER BY` (like you had originally suggested in the issue).

Regarding performance of traversing an index in reverse order. I believe that you're correct here. The query planner will start by requesting the last key and then call previous key until the entire index has been scanned. The overhead of this approach versus scanning the index forwards is negligible in this particular case.

I ran this by @jcjones and he was able to confirm this with a sample query in staging.

Reply:
Sounds good, thanks for checking!

Reply:
Confirmed the actual innodb handlers I'm thinking of exist in our MariaDB version:

`index_last`
https://github.com/MariaDB/server/blob/098c0f2634a35e24e9eb0f06d3fba69d0d097657/storage/innobase/handler/ha_innodb.cc#L9279

`index_prev`
https://github.com/MariaDB/server/blob/098c0f2634a35e24e9eb0f06d3fba69d0d097657/storage/innobase/handler/ha_innodb.cc#L9243